### DISCUSSION TITLE
could we edit the supported Ironwolf driver list with some DB file

### DISCUSSION THREAD

**Discussion by vista20sun** [007revad/Synology_HDD_db#410] [Q&A] [answered]
based on the synology official [kb](https://kb.synology.com/DSM/tutorial/Which_Synology_DiskStation_RackStation_supports_Seagate_IronWolf_Health_Management) only the model in the official db can enable ironwolf test, (e.g. ST12000VN0008 - 2JH101 could enable IHM (ironwolf health management) in storage manager, but ST12000VN0008 - 2YS101 cannot have the IHM feature.

Is this check controlled by some db file just like the `disk-compatibility` and we can manully update the list? Search a lot but not find any answer.

**Comment by 007revad** [comment]
Ironwolf Health was only available on Synology NAS released before 2022
https://kb.synology.com/en-global/DSM/tutorial/Which_Synology_DiskStation_RackStation_supports_Seagate_IronWolf_Health_Management

2022 was when Synology's partnership with Seagate ended, and also when Synology launched it's own branded drives.

According to [Seagate](https://www.seagate.com/au/en/products/nas-drives/ironwolf-hard-drive/health/) "Seagate® IronWolf and IronWolf Pro drives of all capacities are IHM-enabled." so all Ironwolf drives still include IHM.

I've compared the DS116's DSM 7.1.1 to DSM 7.2.2 and they both seem to have the same files that contain "IHM". I cannot find any file that contains a list of supported drive model numbers.

The one interesting thing I found was dhm_tool which is a binary file that says it is "SeaDragon_DHMr - Seagate drive utilities". I assume DHM stands for Drive Health Monitor". 

The dhm_tool file contains a hard-coded list of supported models. But it only contains ST12000VN0008 and not ST12000VN0008-2YS101 or ST12000VN0008-2JH101

```
root@DISKSTATION:~# dhm_tool --scan
===============================================================================
 SeaDragon_DHMr - Seagate drive utilities
 Copyright (c) 2016-2019 Seagate Technology LLC, All Rights Reserved
 SeaDragon_DHMr Version: 1.3.3-1_18_3 X86_64
 Build Date: Aug 27 2019
 Today: Sun Dec 29 19:16:38 2024
===============================================================================
Vendor   Handle       Model Number            Serial Number          FwRev
ATA      /dev/sg0     ST16000VN001-2YU101     <redacted>             SC60
ATA      /dev/sg1     ST16000VN001-2YU101     <redacted>             SC60
ATA      /dev/sg2     ST16000VN001-2YU101     <redacted>             SC60
ATA      /dev/sg3     ST16000VN001-2YU101     <redacted>             SC60
```

But it doesn't like my Ironwolf drives.
```
root@DISKSTATION:~# dhm_tool -d /dev/sg0 -t
/dev/sg0 - This drive (ST16000VN001-2YU101) is not a supported Seagate device.
```

**Reply by vista20sun** [reply, marked as answer]
Thanks a lot, that the information I'm looking for. 
Main points is the version of synology's `SeaDragon_DHMr` is too old to support the newest Ironwolf driver in DSM. replace the dhm_tool with newer binary executable file could make DSM support IHM on newer ironwolf devices. if needed, I can share how to get the newer version of `SeaDragon_DHMr`.

p.s.
the version in synology could reconized the 2YS101 as IRONWOLF device but not a supported Segate device
```
This drive (ST12000VN0008-2YS101) is not a supported Seagate device.
This drive (ST4000VN008-2DR166) is a supported Seagate device.
This drive (ST12000VN0008-2YS101) is a Seagate IRONWOLF device.
This drive (ST4000VN008-2DR166) is a Seagate IRONWOLF device.
```

Thanks again for sharing the information.

**Reply by 007revad** [reply]
> if needed, I can share how to get the newer version of `SeaDragon_DHMr`

Yes please. I searched for "SeaDragon_DHMr" and "dhm_tool" and found nothing.

**Reply by vista20sun** [reply]
`SeaDragon_DHMr - Seagate drive utilities` is not a public distrubited per-complied cli tools from Seagate, It distributed with NAS system that supported IronWorf Health management.

Luckly I found this [QNAP document](https://www.qnap.com/en-us/how-to/faq/article/why-doesnt-my-seagate-ironwolf-drive-support-ironwolf-health-management-ihm) and notice the `stx_ihm` is the binary executale file we need.

**Reply by vista20sun** [reply]
-----
> this part is how I got the file, please deleted this reply if you see any concern

## 1 got the binary file
you should be able to extract the binary executable file from any nas support IHM, but different brand may have different alias for the file, so I still chose got the file from QNAP as we known it name from the above document.
1.  download the QNAP Nas system.
    download the [QuTScloud](https://www.qnap.com/en/download?model=qutscloud&category=firmware) firmware. I use VMDK image ang load it into vmware Workstation bridge with bridge network mode. (do not use the frimware for real Nas Model, they need to be decrypt on QNAP Nas). You can choise any type of vm disk image you like
2. open the virtural machine, ssh credentials will be provided on screen when VM ready.
3. login the ssh, use `which` find the path of binary executable file.
4. use `scp` extract the file from vm.
## 2 set up the new binary
6. replace the `/usr/syno/sbin/dmh_tool` with new file. 
7. no reboot needed, open DSM -> storage manager, you can see the IHM tab shows up in the Ironwolf device that not supported by synlogy before.

**Reply by 007revad** [reply]
If your Synology is x86_64 can you attach a zipped copy of your stx_ihm

**Reply by vista20sun** [reply]
[stx_ihm.zip](https://github.com/user-attachments/files/18278010/stx_ihm.zip)

**Reply by 007revad** [reply]
One more favor. Can you provide a copy of Qnap's x86_64 `/bin/PCI`

**Reply by vista20sun** [reply]
Sorry, I cannot. I didn't see  `/bin/PCI` in  the basic firmware of QNAP.

**Reply by vista20sun** [reply]
You probablly mean the `/sbin/PC1`
[PC1.zip](https://github.com/user-attachments/files/18278688/PC1.zip)

**Reply by 007revad** [reply]
Nice. Now I should be able to decrypt and decompile qnap firmware.

**Reply by 007revad** [reply]
[v3.6.109](https://github.com/007revad/Synology_HDD_db/releases/tag/v3.6.109)
- Added -I, --ihm option to update IronWolf Health Management to 2.5.1 to support recent model IronWolf and IronWolf Pro drives.
    - For NAS with x86_64 CPUs only.
    - Installs IronWolf Health Management on '22 series and newer models that don't have IronWolf Health Management (untested).

**Reply by vista20sun** [reply]
Test on 23 series, using following script could enable ihm in UI.
```
synoinfo="/etc.defaults/synoinfo.conf"
support_ihm="support_ihm"
/usr/syno/bin/synosetkeyvalue "$synoinfo" "$support_ihm" "yes"

```

**Reply by 007revad** [reply]
@vista20sun Version 3.6.109 and later versions of syno_hdd_db already set `support_ihm="yes"` if syno_hdd_db is run with the -I or --ihm option.
